source/common/router/config_impl.cc

```diff
@@ -505,7 +505,7 @@ std::string RouteEntryImplBase::newPath(const Http::HeaderMap& headers) const {
   if (!port_redirect_.empty()) {
     final_port = port_redirect_.c_str();
   } else {
-    final_port = "";
```

Comment:
I may be outvoted here, but while I totally support the default of not doing
http://foo.com:80/bar -> https://foo.com:80/bar
I think I'd prefer that we handle that by removing default ports which no longer apply.  So http:80 would have :80 removed on an https redirect, https:443 would have the 443 removed on an http redirect, regardless of if the redirect were done via https_redirect_ or the scheme label.  It's a bit more complicated but I think it would result in more consistent behavior.  What do you think?

Reply:
This sounds reasonable to me.